Parser/GenericParser/detectEncoding.m

```diff
@@ -0,0 +1,139 @@
+function [encoding, machineformat] = detectEncoding(filename)
+%function [encoding, machineformat] = detectEncoding(filename)
+%
+% A function to load/detect the Encoding of a file.
+% The detection is dynamic, but unfortunately can be slow since
+% we need to load the entire file several times.
+%
+% Inputs:
+%
+% filename - a string with the filename path.
+%
+% Outputs:
+%
+% encoding - an encoding string
+% machineformat - the machineformat string
+%
+% Example:
+% filename = 'abc.dat';
+% [encoding,mf] = detectEncoding(filename);
+% assert(strcmp(encoding,'windows-1252'));
+% assert(strcmp(mf,'ieee-le.l64'));
+% % detection
+% filename = 'data/testfiles/RBR/duet3/v000/082533_20190525_0515_eng_rbrduet3.txt'
+% encoding = detectEncoding(filename,true);
+% assert(strcmp(encoding,'windows-1252'));
+%
+% author: hugo.oliveira@utas.edu.au
+%
+
+mflist = machineformat_list();
+mclist = encoding_list();
+
+badstring_list = {'�'};
+
+for m = 1:length(mflist)
+
+    for n = 1:length(mclist)
+        machineformat = mflist{m};
+        encoding = mclist{n};
+        fid = fopen(filename, 'r', machineformat, encoding);
```

Comment:
it would be great to test first that the file already exists.  otherwises, the error message is not really adequate

Reply:
Thanks! I will bail out when fid<0. File checking should be done atop this function but it's good practice.